### PR TITLE
Fix issue with persistent availability error with faulty state

### PR DESCRIPTION
**Pull request by ramynasr**
Related to #

## Proposed Changes

*

## Why are these changes being made?

*

## Testing Instructions

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by leonardost** [comment]
I've talked with @hambai today and #96641 might be preferable because of the async nature of `setState`. Passing a callback to `setState` ensures that it will be run after `setState` updates, while chaining code with `.then` might result in sometimes running code that depends on the TLD list and the list not having been loaded into state yet.

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -1406,6 +1424,9 @@ class RegisterDomainStep extends Component {
 	onSearch = async ( searchQuery ) => {
 		debug( 'onSearch handler was triggered with query', searchQuery );
 
+		// Trigger a clean state when a new search is starting
+		this.onSearchChange( searchQuery );
```
Comment (leonardost) [inline client/components/domains/register-domain-step/index.jsx:1428]: I think this is fine, but I wonder if calling `onSearchChange` every time here would have unintended consequences 🤔 
My suggestion would be removing this call and adding these properties to the `setState` call right below:
```js
				// Clean errors from previous searches
				showAvailabilityNotice: false,
				availabilityError: null,
				availabilityErrorData: null,
				showSuggestionNotice: false,
				suggestionError: null,
				suggestionErrorData: null,
				suggestionErrorDomain: null,
```
That would only clean previous searches' errors.

**Review by leonardost — APPROVED** [review]
Tested and works as expected 👍 left just one suggestion

**Comment by ramynasr** [comment]
We finally deployed another solution for that.

**Pull request closed without merging**